ext/curl/curl_file.c

```diff
@@ -126,3 +127,29 @@ void curlfile_register_class(void)
 	curl_CURLFile_class->serialize = zend_class_serialize_deny;
 	curl_CURLFile_class->unserialize = zend_class_unserialize_deny;
 }
+
+PHP_CURL_API zend_class_entry *curl_CURLStringFile_class;
+
+ZEND_METHOD(CURLStringFile, __construct)
+{
+	zend_string *data, *postname, *mime = NULL;
+	zval *object;
+
+	object = ZEND_THIS;
+
+	ZEND_PARSE_PARAMETERS_START(2,3)
+		Z_PARAM_STR(data)
+		Z_PARAM_STR(postname)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_STR(mime)
+	ZEND_PARSE_PARAMETERS_END();
+
+	zend_update_property_str(curl_CURLStringFile_class, Z_OBJ_P(object), "data", sizeof("data") - 1, data);
+	zend_update_property_string(curl_CURLStringFile_class, Z_OBJ_P(object), "postname", sizeof("postname")-1, ZSTR_VAL(postname));
```

Comment:
is there any specific reason you use both `zend_update_property_string()` here and `zend_update_property_str()`, even though the latter would be sufficient (and wouldn't require a `ZSTR_VAL()` call)?

Reply:
As I mentioned before we need to truncate mime and postname to null-terminated strings. But possibly we can do it later during reading their values. In this case, I do not know how to provide a default value for mime property for method zend_update_property_str . Looks like I required to use "zend_string_init" method, but at least I do not know which value for its third option I must use.

Reply:
You suggested code looks reasonable to me.

For zend_string_init the last parameter should be 0, but in this case it's probably a bit nicer to write it as you did.